pkgs/applications/office/sleek/default.nix

```diff
@@ -0,0 +1,78 @@
+{ fetchFromGitHub
+, mkYarnPackage
+, fetchurl
+, fpm
+, wrapGAppsHook
+, lib
+}:
+
+mkYarnPackage rec {
+  pname = "sleek";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "ransome1";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-j5T4StCcHtNHEn0VycPJlX/w+F0jZYBPEgHI/n5nGiI=";
+  };
+
+  electron = fetchurl {
+    url = "https://github.com/electron/electron/releases/download/v18.3.7/electron-v18.3.7-linux-x64.zip";
+    hash = "sha256-4JjmZB6HWjBZhRiwtFDMKsG8/nfvK/1jkvIxJU+DUos=";
+  };
+
+  nativeBuildInputs = [ wrapGAppsHook ];
+
+  patches = [
+    ./package.json.patch
+    ./snap/gui/sleek.desktop.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace src/scss/style.scss --replace '../../node_modules' $node_modules
+    substituteInPlace package.json --replace 'yarn' 'yarn --offline'
+
+    export HOME=$(mktemp -d)
+
+    mkdir -p $HOME/.cache/{electron,electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86_64}
+    ln -s ${electron} $HOME/.cache/electron/electron-v18.3.7-linux-x64.zip
+    ln -s ${fpm}/bin/fpm $HOME/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86_64/fpm
+  '';
+
+  buildPhase = ''
+    yarn --offline build:linux
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -vp  $out/opt $out/bin $out/share/{icons,applications}
+    cp -r deps/${pname}/dist/linux-unpacked/* $out/opt/
+    ln -s $out/opt/${pname} $out/bin/
```

Comment:
Please do not substitute pname wherever possible

Reply:
I thought it may reduce spelling errors…

Reply:
It's removed now.